src/segmented-control/src/SegmentedControl.js

```diff
@@ -1,144 +1,137 @@
-import React, { PureComponent } from 'react'
+import React, { memo, useState, useEffect } from 'react'
 import PropTypes from 'prop-types'
 import Box, { spacing, position, layout, dimensions } from 'ui-box'
 import safeInvoke from '../../lib/safe-invoke'
 import SegmentedControlRadio from './SegmentedControlRadio'
 
 let radioCount = 1 // Used for generating unique input names
 
-export default class SegmentedControl extends PureComponent {
-  static propTypes = {
-    /**
-     * Composes some Box APIs.
-     */
-    ...spacing.propTypes,
-    ...position.propTypes,
-    ...layout.propTypes,
-    ...dimensions.propTypes,
-
-    /**
-     * The options for the radios of the Segmented Control.
-     */
-    options: PropTypes.arrayOf(
-      PropTypes.shape({
-        label: PropTypes.node.isRequired,
-        value: PropTypes.oneOfType([
-          PropTypes.number,
-          PropTypes.string,
-          PropTypes.bool
-        ]).isRequired
-      })
-    ).isRequired,
-
-    /**
-     * The current value of the Segmented Control when controlled.
-     */
-    value: PropTypes.oneOfType([
-      PropTypes.number,
-      PropTypes.string,
-      PropTypes.bool
-    ]),
-
-    /**
-     * The default value of the Segmented Control when uncontrolled.
-     */
-    defaultValue: PropTypes.oneOfType([
-      PropTypes.number,
-      PropTypes.string,
-      PropTypes.bool
-    ]),
-
-    /**
-     * Function called when the value changes.
-     */
-    onChange: PropTypes.func,
-
-    /**
-     * The name of the radio group.
-     */
-    name: PropTypes.string,
-
-    /**
-     * The height of the Segmented Control.
-     */
-    height: PropTypes.number,
-
-    /**
-     * When true, the Segmented Control is disabled.
-     */
-    disabled: PropTypes.bool
+const SegmentedControl = memo(props => {
+  const {
+    value,
+    name,
+    height = 32,
+    options,
+    onChange,
+    defaultValue,
+    disabled,
+    ...rest
+  } = props
+
+  const groupName = `SegmentedControl-${radioCount}`
+  radioCount += 1
```

Comment:
This will regen the groupName and increment the counter on every render. Check out this for a good example of solving for this so it only ever increments on _mount_:

https://sourcegraph.segment.com/github.com/segmentio/evergreen@v5/-/blob/src/text-input/src/TextInputField.js?subtree=true#L11:66